Add V3.0.2 Changes to 3.1.0 (#254)
admin-shell-io/aas-specs-api: documentation/IDTA-01002-3/modules/ROOT/pages/IDTA-01002_HTTP-REST-API.adoc

The following constraints apply for the combination of modifiers:

> * For Content=Value, the requested object is always serialized to an unnamed JSON Object or Array. This means that the response object must not have with the object’s idShort at the root level.

Reviewer: ```suggestion
* For Content=Value, the requested object is always serialized to an unnamed JSON Object or Array. This means that the response object must not have a property with the object’s idShort at the root level.
```

> * For Content=Value, the requested object is always serialized to an unnamed JSON Object or Array. This means that the response object must not have a property with the object’s idShort at the root level.

Reviewer 2: ```suggestion
* For Content=Value, the requested object shall always be serialized to an unnamed JSON Object or Array. This means that the response object must not have a property with the object’s idShort at the root level.
```

* If Level=Core and Content=Value, only the requested object and the direct children without their value (empty value) will be returned in value serialization. If a direct child is a SubmodelElementCollection, "<SubmodelElementCollection/idShort>": \{} will be returned. If a direct child is a SubmodelElementList, "<SubmodelElementList/idShort>": [] will be returned.
* The combination of Content=Metadata and Extent=WithBLOBValue is not allowed.

> * The behavior of Level in combination with Content=Metadata is undefined. A server may ignore the Level modifier if it is set.

Reviewer: What is the alternative to ignoring it? Returning `404 - Not Found` is not allow, right? And actually applying the level modifier is also not possible. 
This statement should clearly list the possible allowed behavior.

Author: What's your suggestion? `400`? `404` doesn't seem correct to me, as the resource in general exists at the specified location.

Reviewer: Well that depends on you understanding of what is a breaking change...

**TL;DR**
Formulate it similar to all other constraints, i.e. _The combination of Content=Metadata and Level is not allowed_

**Detailed Answer**
Right now, the specification is contradictory saying that you can combine metadata and level on the one hand, and on the other hand saying that for metadata serialization there are no properties like _values_ or _submodelElements_.
Therefore I would argue that we can more or less make any change we like in a bugfix version as it will never break a client implementation as they currently cannot know what to expect.
Following that reasoning, I would treat it like any similar constraint, e.g. the one above in the text (The combination of Content=Metadata and Extent=WithBLOBValue is not allowed) and just formulate it the same way.
However, this still leaves room for interpretation what _not allowed_ means. I would argue that it heavily implies returning `400 Bad Request`. But I would not mention this only for this specific case but either on a higher level (applying to all constraints on modifier combinations) or not at all.

Reviewer 2: We should use terms conformant to https://rwth-iat.github.io/aas-specs-antora/IDTA-01001/snapshot/IDTA-01001_Preamble.html#_conformance
i.e. for example:
If parameter Content is set to "Metadata" then Level shall not be defined.

|GetAssetAdministrationShell |GET |/aas |content-suffix: $reference
|PutAssetAdministrationShell |PUT | |
|GetAllSubmodelReferences |GET |/aas/submodel-refs |Pagination

> |PostSubmodelReference |POST |/aas/submodel-refs | Location header of the response contains the value ‘<baseUrl>/aas/submodel-ref/{submodelIdentifier}’ 

Reviewer 2: ```suggestion
|PostSubmodelReference |POST |/aas/submodel-refs a| Location header of the response contains the value ‘<baseUrl>/aas/submodel-ref/{submodelIdentifier}’ 
```
